finetune/encoding/target_encoders.py

```diff
@@ -181,12 +185,84 @@ def target_labels(self):
         raise ValueError
 
 
-class SequenceLabelingEncoder(LabelEncoder, BaseEncoder):
-    pass
+class SequenceLabelingEncoder(BaseEncoder):
+
+    def __init__(self, pad_token="<PAD>"):
+        self.classes_ = None
+        self.pad_token = pad_token
+        self.lookup = None
+
+    def fit(self, labels):
+        self.classes_ = list(set(lab["label"] for lab in labels) | {self.pad_token})
+        self.lookup = {c: i for i, c in enumerate(self.classes_)}
+
+    def pre_process_label(self, out, labels):
+        # TODO employ more complex strategies
+        pad_idx = self.lookup[self.pad_token]
+        return labels, pad_idx
+
+    @staticmethod
+    def overlaps(label, tok_start, tok_end, tok_text):
+        does_overlap = (
+            label["start"] < tok_end <= label["end"] or
```

Comment:
I think as currently written this requires annotations to be >1 char? So we couldn't annotate `$` as currency, for instance?